### PR TITLE
Update compositingAlphaMode -> alphaMode

### PR DESCRIPTION
**Pull request by shrekshao**
Issue: #1561

Split from https://github.com/gpuweb/cts/pull/1657

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/web_platform/copyToTexture/canvas.spec.ts [hunk 1]
```diff
@@ -352,7 +352,7 @@ class F extends CopyToTextureUtils {
 
     // The source canvas has bgra8unorm back resource. We
     // swizzle the channels to align with 2d/webgl canvas and
-    // clear alpha to opaque when context compositingAlphaMode
+    // clear alpha to opaque when context alphaMode
     // is set to opaque (follow webgpu spec).
```
Comment (kainino0x) [inline src/webgpu/web_platform/copyToTexture/canvas.spec.ts:356]: compositingAlphaMode wasn't supposed to affect the results of readbacks. I'm unclear about what this comment was talking about when it says "follow webgpu spec".

It seems like these functions are passing through a `premultiplied: boolean` when what they really mean is `alphaMode: GPUCanvasAlphaMode`, but I'm not sure about that. I understood this when I reviewed it but I don't anymore.

Can you investigate and maybe clarify the comments and function parameters where it makes sense?

Comment (shrekshao) [reply src/webgpu/web_platform/copyToTexture/canvas.spec.ts:359]: Change `premultiplied: boolean` to `alphaMode: GPUCanvasAlphaMode` for gpu context canvas. (Preserve those for 2d, webgl, and related utils)

**Review by kainino0x — COMMENTED** [review]
Sorry for the delay on this. It was kind of hard to understand and I kept putting it off. I finally checked out the code and did a bunch of renames/refactors until I understood what was going on here.

Everything checks out except I think there's one thing that can be made more robust. I left a `// TODO`. If you have a chance and these tests should be passing in the latest Chromium, can you make sure these tests are passing, then try and fix the TODO and see if they're still passing?

**Comment by shrekshao** [comment]
Thanks for the review and change. Yeah turns out my previous refactor was not thorough.
Having `colorValue = alphaValue` do pass in latest Chrome canary.

File: src/webgpu/web_platform/copyToTexture/canvas.spec.ts [hunk 2]
```diff
@@ -237,15 +237,9 @@ class F extends CopyToTextureUtils {
     const rectHeight = Math.floor(height / 2);
 
     const alphaValue = 153;
-    // TODO: I think colorValue can just always = alphaValue now, because readback will clear it to
+    // colorValue can just always = alphaValue now, because readback will clear it to
     // 255 if opaque. Need to try this and verify it.
```
Comment (kainino0x) [inline src/webgpu/web_platform/copyToTexture/canvas.spec.ts]: Did you verify it works as expected in Chrome? If so, update this comment.

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/web_platform/copyToTexture/canvas.spec.ts (hunk 2 above)
Comment (kainino0x) [reply src/webgpu/web_platform/copyToTexture/canvas.spec.ts]: Never mind, I see you said it works.

File: src/webgpu/web_platform/copyToTexture/canvas.spec.ts [hunk 3]
```diff
@@ -215,15 +222,24 @@ class F extends CopyToTextureUtils {
...
+    height: number,
+    alphaMode: GPUCanvasAlphaMode
+  ): Uint8ClampedArray {
     const rectWidth = Math.floor(width / 2);
     const rectHeight = Math.floor(height / 2);
 
     const alphaValue = 153;
-    const colorValue = premultiplied ? alphaValue : 255;
+    // colorValue can just always = alphaValue now, because readback will clear it to
+    // 255 if opaque.
```
Comment (kainino0x) [inline src/webgpu/web_platform/copyToTexture/canvas.spec.ts:241]: I think I misunderstood a little bit when I was writing this comment, but anyway it's not entirely relevant anymore. Here's a more clarifying comment, if you think it makes sense:

```suggestion
    // Always output [153, 153, 153, 153]. When the alphaMode is...
    //   - premultiplied: the readback is CSS `rgba(255, 255, 255, 60%)`.
    //   - opaque: the readback is CSS `rgba(153, 153, 153, 100%)`.
    // getExpectedReadbackForWebGPUCanvas matches this.
```

**Review by kainino0x — APPROVED** [review]

**Pull request merged**